### PR TITLE
Nc news 11

### PR DESCRIPTION
**Pull request by XingWei0920**
I finish task 11

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -361,4 +360,58 @@ describe("/api/articles/:article_id/comments", () => {
...
+
+  test("GET:404 sends an Bad Request error if the article_id is not a number", () => {
+    return request(app)
+      .get("/api/articles/banana")
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe("Bad Request");
+      });
+  });
+
```
Comment (nc-jonathan-mcguire) [inline __tests__/app.test.js]: What about a topic that has no articles i.e. paper?

File: controllers/server-controllers.js [hunk 2]
```diff
@@ -44,12 +44,29 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+            .catch(next)
+        }
+        else
+        {
+            selectArticles()
+            .then((articles)=>{
+               res.status(200).send({articles});
+           })
+           .catch(next)
+        }
```
Comment (nc-jonathan-mcguire) [inline controllers/server-controllers.js]: Not sure you need this if statement, seem at though they both call the same function

**Review by nc-jonathan-mcguire — CHANGES_REQUESTED** [review]
We need a test and please see my comment about your if statement

- [ ] Status 200: OK Responds with an empty array when topic exists but has no associated articles. You can use `paper` to test this

File: models/server-models.js [hunk 3]
```diff
@@ -32,19 +32,23 @@ exports.selectArticles = (topic) => {
...
             else
             {
                 return result.rows;
             }
             
-          });
+               });
+        
+            })
     }
```
Comment (nc-mick-fay) [inline models/server-models.js:45]: This if/else isn't necessary if you are only add your topic into your query string if topic exists.

your check to see if rows.length === 0 isn't needed either - rows in this case is an empty array, so you can just return result.rows like you are doing in your else statement.

**Review by nc-mick-fay — APPROVED** [review]
Good changes made from the previous commit, and a nice additional test. 

Some of the logic in your model isn't really necessary. You could set up your query string to only add a topic if you pass one in so you don't need to have two separate queries. You may want to consider how you would refactor this is in the future.

I'm happy to approve this pull request - merge into main when you're ready

**Pull request merged**